### PR TITLE
Move metric & segment based GUI query editor to Admin

### PR DESCRIPTION
**Pull request by ranquild**
This is to simplify search for usages of legacy query. This change also surfaced non-obvious legacy query dependencies.

### CODE REVIEW

File: frontend/src/metabase/admin/datamodel/components/FilterList/FilterList.jsx [hunk 1]
```diff
@@ -3,14 +3,14 @@ import { Component } from "react";
 import { connect } from "react-redux";
 
 import { getMetadata } from "metabase/selectors/metadata";
-import Filter from "./Filter";
-import { filterWidgetFilterRenderer } from "./filters/FilterWidget";
+import { filterWidgetFilterRenderer } from "../FilterWidget/FilterWidget";
```
Comment (kamilmielnik) [inline frontend/src/metabase/admin/datamodel/components/FilterList/FilterList.jsx:6]:
```suggestion
import { filterWidgetFilterRenderer } from "../FilterWidget";
```

File: frontend/src/metabase/admin/datamodel/components/GuiQueryEditor/GuiQueryEditor.jsx [hunk 2]
```diff
@@ -10,12 +10,15 @@ import { Icon } from "metabase/core/components/Icon";
...
 import { DatabaseSchemaAndTableDataSelector } from "metabase/query_builder/components/DataSelector";
-import AggregationWidget from "./AggregationWidget";
-import BreakoutWidget from "./BreakoutWidget";
-import FilterWidgetList from "./filters/FilterWidgetList";
-import { FilterPopover } from "./filters/FilterPopover";
-
-export default class GuiQueryEditor extends Component {
+import { AggregationWidget } from "../AggregationWidget";
+import { BreakoutWidget } from "../BreakoutWidget";
+import { FilterWidgetList } from "../FilterWidgetList/FilterWidgetList";
```
Comment (kamilmielnik) [inline frontend/src/metabase/admin/datamodel/components/GuiQueryEditor/GuiQueryEditor.jsx:15]:
```suggestion
import { FilterWidgetList } from "../FilterWidgetList";
```

File: frontend/src/metabase/admin/datamodel/metadata/components/FieldRemappingSettings/FieldRemappingSettings.styled.tsx [hunk 3]
```diff
@@ -1,7 +1,7 @@
 import styled from "@emotion/styled";
 import { alpha, color } from "metabase/lib/colors";
 import SelectButton from "metabase/core/components/SelectButton";
-import FieldList from "metabase/query_builder/components/FieldList";
+import { FieldList } from "metabase/admin/datamodel/components/FieldList/FieldList";
```
Comment (kamilmielnik) [inline frontend/src/metabase/admin/datamodel/metadata/components/FieldRemappingSettings/FieldRemappingSettings.styled.tsx:4]:
```suggestion
import { FieldList } from "metabase/admin/datamodel/components/FieldList";
```

File: frontend/src/metabase/components/DateAllOptionsWidget/DateAllOptionsWidget.tsx [hunk 4]
```diff
@@ -2,7 +2,7 @@ import { useState } from "react";
 import { t } from "ttag";
 import cx from "classnames";
 
-import DatePicker from "metabase/query_builder/components/filters/pickers/DatePicker/DatePicker";
+import DatePicker from "metabase/admin/datamodel/components/filters/pickers/DatePicker/DatePicker";
```
Comment (kamilmielnik) [inline frontend/src/metabase/components/DateAllOptionsWidget/DateAllOptionsWidget.tsx:5]: It'd be nice to add named exports here.

```suggestion
import DatePicker from "metabase/admin/datamodel/components/filters/pickers/DatePicker";
```

File: frontend/src/metabase/dashboard/containers/AutomaticDashboardApp.jsx [hunk 5]
```diff
@@ -14,7 +14,7 @@ import ActionButton from "metabase/components/ActionButton";
 import Button from "metabase/core/components/Button";
 import Card from "metabase/components/Card";
 import { Icon } from "metabase/core/components/Icon";
-import Filter from "metabase/query_builder/components/Filter";
+import { Filter } from "metabase/admin/datamodel/components/Filter/Filter";
```
Comment (kamilmielnik) [inline frontend/src/metabase/dashboard/containers/AutomaticDashboardApp.jsx:17]:
```suggestion
import { Filter } from "metabase/admin/datamodel/components/Filter";
```

File: frontend/src/metabase/parameters/utils/date-formatting.ts [hunk 6]
```diff
@@ -3,8 +3,8 @@ import _ from "underscore";
 // eslint-disable-next-line no-restricted-imports -- deprecated usage
 import moment from "moment-timezone";
 
-import { DATE_OPERATORS } from "metabase/query_builder/components/filters/pickers/DatePicker/DatePicker";
-import { EXCLUDE_OPERATORS } from "metabase/query_builder/components/filters/pickers/DatePicker/ExcludeDatePicker";
+import { DATE_OPERATORS } from "metabase/admin/datamodel/components/filters/pickers/DatePicker/DatePicker";
+import { EXCLUDE_OPERATORS } from "metabase/admin/datamodel/components/filters/pickers/DatePicker/ExcludeDatePicker";
```
Comment (kamilmielnik) [inline frontend/src/metabase/parameters/utils/date-formatting.ts:7]: It'd be nice to add named exports here.

```suggestion
import {
  DATE_OPERATORS,
  EXCLUDE_OPERATORS
} from "metabase/admin/datamodel/components/filters/pickers/DatePicker";
```

File: frontend/src/metabase/search/components/SearchFilterDatePicker/SearchFilterDatePicker.tsx [hunk 7]
```diff
@@ -2,9 +2,9 @@ import { useState } from "react";
 import { t } from "ttag";
 import { SearchFilterApplyButton } from "metabase/search/components/SearchFilterPopoverWrapper/SearchFilterPopoverWrapper";
 import { filterToUrlEncoded } from "metabase/parameters/utils/date-formatting";
-import DatePicker from "metabase/query_builder/components/filters/pickers/DatePicker/DatePicker";
-import type { DateShortcutOptions } from "metabase/query_builder/components/filters/pickers/DatePicker/DatePickerShortcutOptions";
-import { DATE_SHORTCUT_OPTIONS } from "metabase/query_builder/components/filters/pickers/DatePicker/DatePickerShortcutOptions";
+import DatePicker from "metabase/admin/datamodel/components/filters/pickers/DatePicker/DatePicker";
```
Comment (kamilmielnik) [inline frontend/src/metabase/search/components/SearchFilterDatePicker/SearchFilterDatePicker.tsx:5]: It'd be nice to add named exports here.

```suggestion
import DatePicker from "metabase/admin/datamodel/components/filters/pickers/DatePicker";
```

**Review by kamilmielnik — APPROVED** [review]
:pray:

**Review by uladzimirdev — APPROVED** [review]

**Pull request merged**